lib/plugin-api/hydrogen-provider.js

```diff
@@ -51,4 +52,28 @@ export default class HydrogenProvider {
 
     return store.kernel.getPluginWrapper();
   }
+
+  /*
+   * Get the `atom$Range` that will run if `hydrogen:run-cell` is called.
+   * `null` is returned if no active text editor.
+   * @return {Class} `atom$Range`
+   */
+  getCellRange() {
+    if (!store.editor) return null;
+    return getCurrentCell(store.editor);
+  }
+
+  /*
+   * Get the `atom$Range` that will run if `hydrogen:run` is called.
+   * `null` is returned if no active text editor.
+   * @return {Class} `atom$Range`
+   */
+  getCodeBlockRange() {
+    if (!store.editor) return null;
+    return findCodeBlock(store.editor);
+  }
+
```

Comment:
Whoops, that's not what this does!

I might just do the cell part for now and coming back to this part later.